### PR TITLE
⚗️ [RUM-3208] Experiment private action name

### PR DESCRIPTION
**Pull request by amortemousque**
## Motivation

Experiment computing a CSS selector used for private action name

## Changes

- Add 3 optional strategies to  [getSelectorFromElement](https://github.com/DataDog/browser-sdk/blob/a74b2684045890e8aa18eb298f86f3f1c228f4de/packages/rum-core/src/domain/getSelectorFromElement.ts#L62) :
  - `stopRecurringWhenUnique`
  - `targetAtomicElement`
  - `onlyPrefixWithSemanticTag`
- Add [getPrivateActionNameFromElement](https://github.com/DataDog/browser-sdk/blob/a74b2684045890e8aa18eb298f86f3f1c228f4de/packages/rum-core/src/domain/action/getActionNameFromElement.ts#L21) which:
  - first try to get the name programmatically
  - then fallback to a CSS selector 
- Use the new strategies to collect `_dd.action.private_name`
- Add [measurePerformance](https://github.com/DataDog/browser-sdk/blob/a74b2684045890e8aa18eb298f86f3f1c228f4de/packages/core/src/tools/measurePerformance.ts#L4) to measure the action name and selector computation duration and collect them in:
```
_dd.action.performance: { 
  selector_duration: Duration | undefined
  name_duration: Duration | undefined
  private_name_duration: Duration | undefined
}
```

## Testing

- [x] Local
- [x] Staging
- [x] Unit
- [ ] End to end

---

I have gone over the [contributing](https://github.com/DataDog/browser-sdk/blob/main/CONTRIBUTING.md) documentation.

### CODE REVIEW

File: packages/rum-core/src/domain/getSelectorFromElement.ts [hunk 1]
```diff
@@ -38,16 +55,37 @@ const UNIQUE_AMONG_CHILDREN_SELECTOR_GETTERS: SelectorGetter[] = [
...
   let targetElementSelector = ''
   let element: Element | null = targetElement
+  const prefix = onlyPrefixWithSemanticTag ? getSemanticTagName : getTagNameSelector
+
+  if (targetMeaningfulElement && targetElement.closest) {
+    const closestClickableElement = targetElement.closest(MEANINGFUL_ELEMENTS_SELECTORS.join(','))
+    if (closestClickableElement) {
+      element = closestClickableElement
+    }
+  }
```
Comment (bcaudan) [inline packages/rum-core/src/domain/getSelectorFromElement.ts:72]: **💬 suggestion:** could we stick to either ‏clickable or meaningful naming? or is it worth differentiating the two?

Comment (amortemousque) [reply packages/rum-core/src/domain/getSelectorFromElement.ts]: Nop no reason to differentiating the two

File: packages/rum-core/src/domain/getSelectorFromElement.spec.ts [hunk 2]
```diff
@@ -157,10 +158,73 @@ describe('getSelectorFromElement', () => {
...
+      targetMeaningfulElement?: boolean
+      onlyPrefixWithSemanticTag?: boolean
+    },
+    actionNameAttribute?: string
+  ): string {
     return getSelectorFromElement(
       typeof htmlOrElement === 'string' ? appendElement(htmlOrElement) : htmlOrElement,
-      actionNameAttribute
+      actionNameAttribute,
+      options
```
[the remark this answers is not in the document]
Comment (amortemousque) [reply packages/rum-core/src/domain/getSelectorFromElement.spec.ts]: I go for "not wrapping the tested function at all"

File: packages/rum-core/src/domain/action/trackClickActions.ts [hunk 3]
```diff
@@ -217,6 +220,14 @@ function computeClickActionBase(event: MouseEventOnElement, actionNameAttribute?
       width: Math.round(rect.width),
       height: Math.round(rect.height),
       selector: getSelectorFromElement(event.target, actionNameAttribute),
+      selector_for_private_name:
+        isExperimentalFeatureEnabled(ExperimentalFeature.SELECTOR_FOR_PRIVATE_ACTION_NAME) && !isIE()
```
[the remark this answers is not in the document]
Comment (amortemousque) [reply packages/rum-core/src/domain/action/trackClickActions.ts]: I've added a 'closestElement' polyfill which could benefit in other places but for now I only use it for private names
https://github.com/DataDog/browser-sdk/blob/7f4129dc27e658e8f59a5169edf82fcc8cd0f0bd/packages/core/src/tools/utils/polyfills.ts#L126

**Pull request closed without merging**